clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -0,0 +1,248 @@
+				if SSHToMasterSupported {
+					a.updateMasterAvailabilityMetrics(k8sClient, config)
+				}
+				return
+			}
+
+		}
+	}()
+	return nil, nil
+}
+
+func (a *apiAvailabilityMeasurement) updateAvailabilityMetrics(apiServerAvailable bool, metrics *apiAvailabilityMetrics) {
+	if apiServerAvailable {
+		metrics.updateSuccessMetrics()
+	} else {
+		metrics.updateFailureMetrics()
+	}
+}
+
+// If the consecutive failed probes counter value > 0, then it means we are in the middle of an unavailability period
+// for the api server
+func (a *apiAvailabilityMetrics) previousProbeFailure() bool {
+	return a.consecutiveFailedProbes > 0
+}
+
+// Check if the recently concluded consecutive failed number of probes is
+// higher than the max consecutive failed number of probes so far
+// if yes, then Update max consecutive failed probes
+func (a *apiAvailabilityMetrics) mayBeUpdateMaxConsecutiveFailures() {
+	if a.consecutiveFailedProbes > a.maxConsecutiveFailedProbes {
+		a.maxConsecutiveFailedProbes = a.consecutiveFailedProbes
+	}
+}
+
+func (a *apiAvailabilityMetrics) updateFailureMetrics() {
+	// increment total number of failed probes count, current number of failed probes by 1
+	a.numFailures++
+	a.consecutiveFailedProbes++
+}
+
+func (a *apiAvailabilityMetrics) updateSuccessMetrics() {
+	// if the probe succeeded, then increase the number of successful probes count by 1
+	a.numSuccess++
+	// if this successful probe was received after failing probe(s)
+	if a.previousProbeFailure() {
+		a.mayBeUpdateMaxConsecutiveFailures()
+		// reset current consecutive failed number of probes
+		a.consecutiveFailedProbes = 0
+	}
+}
+
+// Execute starts the api-server healthz probe end point from start action and
+// collects availability metrics in gather.
+func (a *apiAvailabilityMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+
+	SSHToMasterSupported := config.ClusterFramework.GetClusterConfig().SSHToMasterSupported
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+
+	probeFrequency, err := util.GetIntOrDefault(config.Params, "frequency", 1)
+	if err != nil {
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		if a.isRunning {
+			klog.Infof("%s: measurement already running", a)
+			return nil, nil
+		}
+		return a.start(config, SSHToMasterSupported, probeFrequency)
+	case "gather":
+		a.stop()
+		var summaries []measurement.Summary
+		summaries = a.createClusterAvailabilitySummary(probeFrequency, summaries)
+		if SSHToMasterSupported {
+			summaries = a.createMasterAvailabilitySummary(probeFrequency, summaries)
+		}
+		return summaries, nil
+	default:
+		return nil, fmt.Errorf("unknown action %v", action)
+	}
+}
+
+func (a *apiAvailabilityMeasurement) createClusterAvailabilitySummary(probeFrequency int, summaries []measurement.Summary) []measurement.Summary {
+	a.clusterLevelMetrics.mayBeUpdateMaxConsecutiveFailures()
```

Comment:
let's not do this here - instead let's make it a defer in stop() function (or not a defer - just done after wg.Wait()